app/lib/submission_builder/ty2024/states/md/md_return_xml.rb

```diff
@@ -93,6 +94,13 @@ def financial_transaction
               kwargs: { refund_amount: calculator.refund_or_owed_amount }
             ).document.at("*")
           end
+
+          def form_has_non_zero_amounts(form_prefix, calculated_fields)
+            lines = calculated_fields.keys.select { |line_name| line_name.match?(/^"#{form_prefix}"/) }
+            lines.any? do |line_num|
+              calculated_fields.fetch(line_num) != 0
```

Comment:
if the `calculated_fields.fetch(line_num)` returns `nil`, this will also evaluate to true, right? Is that a situation that we don't expect?

Reply:
oh that 's a really great point. should i:
`calculated_fields.fetch(line_num).present? && calculated_fields.fetch(line_num) != 0`
or
`calculated_fields.fetch(line_num).to_i != 0`
i'm leaning towards the first option just in case there's a value like 0.05 or something stupid

Reply:
first option sounds good due to the case you mentioned 👍